editor/editor_themes.cpp

```diff
@@ -351,9 +351,8 @@ Ref<Theme> create_editor_theme(const Ref<Theme> p_theme) {
 	style_popup->set_default_margin(MARGIN_BOTTOM, default_margin_size * 2);
 	style_popup->set_border_color_all(contrast_color_1);
 	style_popup->set_border_width_all(MAX(EDSCALE, border_width));
-	const Color shadow_color = Color(0, 0, 0, dark_theme ? 0.3 : 0.1);
-	style_popup->set_shadow_color(shadow_color);
-	style_popup->set_shadow_size(4 * EDSCALE);
```

Comment:
this is needed, please dont remove

Reply:
@djrm I the shadow for some days now. see if I could get used to is. but since the panel also has a border and the shadow is so small it just feels like a second border which is too much.

Also the shadow makes it feels less connected to the button where the toolbox appears from.

They do make a lot of sense when separating the popup from the rest of the ui. (like for confirmation dialogs)
But in this case the popup is part of the button, so it should not have a hover effect.

I think the best approach actually might be to give the pressed button the same color than the popup so it feels like they were connected.

Reply:
@toger5 you are not considering the whole picture here, popups are used in several other places not just popup buttons, for example, color picker and context menu. They help a lot to make things more visible, if they are flat they simply dont popup. also i dont see much argument in saying that they are should be connected to the button i mean it would be nice, but not the priority.

also i completely disagree that they look like a second border. in any case we could simply increase the shadow not remove it.

![screenshot from 2017-09-28 11-33-05](https://user-images.githubusercontent.com/1103897/30978717-fb02eb52-a440-11e7-8ce1-c110b28a596e.png)
![screenshot from 2017-09-28 11-33-22](https://user-images.githubusercontent.com/1103897/30978720-fc96da6e-a440-11e7-993c-b3805880c1ac.png)

![screenshot from 2017-09-28 11-35-06](https://user-images.githubusercontent.com/1103897/30978787-37a6bd9a-a441-11e7-9b84-74eb9940ef18.png)
![screenshot from 2017-09-28 11-35-38](https://user-images.githubusercontent.com/1103897/30978790-38f56520-a441-11e7-85d8-cbaba1c547ea.png)

Reply:
But in any example they you posted they still are connected to something. be it the vector input part of the inspector, or the view button on the top left of the 3d viewport.
so bringing them on a new layer just feels weird. I also haven't seen any other application doing that.
Increasing the shadow would result in them looking like they would hover even higher which makes it worse imo.

Reply:
@toger5 i simply dont see the reasoning behind wanting to have them connected besides being some sort of theoretical design thing.

Reply:
I don't see the reason to put them further apart by adding a hover effect. since they are already well distinguished by the border.

especially since the rest of the editor is so flat. for the actual popups (alerts, file dialog and so on) it makes sense since they are almost another window. 

especially in situations like this it is not desired to have a smooth transition or a hover effect to move the selectable options away from the ui element.

<img width="231" alt="screen shot 2017-09-28 at 19 10 17" src="https://user-images.githubusercontent.com/16718859/30980253-b1a396c4-a480-11e7-8a4b-2dcf52c80fc5.png">

@djrm I'm actually really sorry that I challenge your nerves so much ;D 

But we don't have to decide today. So we both can think about it a little more and come to an conclusion another day.

Reply:
@toger5 for that case in particular you could change the button code to show the popup behind parent, that could work nice.

Reply:
@djrm oh. nice idea!
Can I also make the shadow more suddle (by a little bit)?

Reply:
@toger5 also, its ok to have discussions, i dont get angry or anything, there is also no reason to think that i should be right, this is one of the things i hate about design stuff, that i cant prove that im correct, since when it comes to design everyone is correct and everyone has an opinion.

Reply:
@toger5 yeah, you can make it more subtle but just for white themes, i think i had a condition for that.

Reply:
@djrm yea really good point of view...

I just noticed something:
![](https://user-images.githubusercontent.com/16718859/30352084-8ee92b3a-981e-11e7-99a8-68311bd13ce5.png)

if you look really carefully they also have a shodow. it's really soft though.
Maybe if we make the buttons draw on top this could work really well too. I also think that we could remove the forced border, when setting border width to 0, for the popups when having shadows which is an advantage because I don't use the borders anyways so it doesn't look like a double border to me ;)

Reply:
@toger5 you could also make the button itself have a shadow, so everything looks raised

Reply:
@djrm but than we definitely get one awkward overlapping shadow...
Or we add an option to have different shadow size for each border (but that is kind of hard... we could do it though)

Reply:
![image](https://user-images.githubusercontent.com/8281454/30982023-3034d1fc-a4c1-11e7-8574-de560dbeba2d.png)
if you make popup menu behind the button, you need to consider this kind of case.

Reply:
@volzhs yeah thats true, @toger5  the problem here is that those menus dont necesarily match the button position.

Reply:
Bump, what's the status?

Reply:
Anyone still around? :)

Reply:
im not sure if toger still wants to push this, i personally dont see much point in this change and some concerns are still unresolved.

Reply:
I still would like it to get merged. But it is not really important.

@akien-mga I'm currently pretty busy. I think you just want the pr to be closed. so If I haven't updated the shadow for the small popups in a couple (about 3 days) you can also close

Reply:
or if @djrm is okay with removing them ;D